File: rock-paper-scissors/typescript/src/rockPaperScissors.ts
export type Play = 'rock' | 'paper' | 'scissors';
export type Outcome = 'win' | 'lose' | 'draw';

const beats: Record<Play, Play> = {
  rock: 'scissors',
  scissors: 'paper',
  paper: 'rock',
};

export function decide(p1: Play, p2: Play): Outcome {
  if (p1 === p2) return 'draw';
  return beats[p1] === p2 ? 'win' : 'lose';
}

File: rock-paper-scissors/python/src/rock_paper_scissors/rock_paper_scissors.py
from enum import StrEnum


class Play(StrEnum):
    ROCK = "rock"
    PAPER = "paper"
    SCISSORS = "scissors"


class Outcome(StrEnum):
    WIN = "win"
    LOSE = "lose"
    DRAW = "draw"


_BEATS: dict[Play, Play] = {
    Play.ROCK: Play.SCISSORS,
    Play.SCISSORS: Play.PAPER,
    Play.PAPER: Play.ROCK,
}


def decide(p1: Play, p2: Play) -> Outcome:
    if p1 == p2:
        return Outcome.DRAW
    return Outcome.WIN if _BEATS[p1] == p2 else Outcome.LOSE
